Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -3889,6 +3889,54 @@ static NSStringCompareOptions compareOptionsForFindOptions(WebCore::FindOptions
 
 #if PLATFORM(IOS_FAMILY)
 
+std::optional<FloatRect> UnifiedPDFPlugin::highlightRectForTapAtPoint(FloatPoint pointInRootView) const
+{
+    RetainPtr annotation = annotationForRootViewPoint(roundedIntPoint(pointInRootView));
+    if (!annotation)
+        return std::nullopt;
+
+    if (!annotationIsLinkWithDestination(annotation.get()))
+        return std::nullopt;
+
+    return { pageToRootView([annotation bounds], [annotation page]) };
+}
+
+void UnifiedPDFPlugin::handleSyntheticClick(PlatformMouseEvent&& event)
+{
+#if HAVE(PDFDOCUMENT_SELECTION_WITH_GRANULARITY)
+    auto pointInRootView = event.position();
+    RetainPtr selection = m_currentSelection;
+    if (selection && event.shiftKey()) {
```

Comment:
Should we allow fallthrough here even with an active selection? Currently we do navigate when shift-clicking on a link annotation while a selection is active

Reply:
Makes sense — I can make that adjustment in my next PR...